src/client/Client.cc

```diff
@@ -6083,8 +6083,21 @@ int Client::_lookup(Inode *dir, const string& dname, int mask, InodeRef *target,
   }
 
   if (dname == "..") {
-    if (dir->dentries.empty())
+    if (dir->dentries.empty()) {
+      MetaRequest *req = new MetaRequest(CEPH_MDS_OP_LOOKUPPARENT);
+      filepath path(dir->ino);
+      req->set_filepath(path);
+
+      InodeRef tmptarget;
+      int r = make_request(req, perms, &tmptarget, NULL, rand() % mdsmap->get_num_in_mds());
+
+      if (r == 0) {
+	dir = tmptarget.get();
+	_ll_get(dir);
+	ldout(cct, 3) << __func__ << " found dir " << (dir)->ino << dendl;
```

Comment:
Is it ok to clobber "dir" here? It does get used at the end of the function so this might subtly change the ldout messages at the end. Maybe just go ahead and store it in *target and leave "dir" alone here?

Reply:
@ukernel @jtlayton Done. Please check now.